tsl/src/hypercore/arrow_cache_explain.c

```diff
@@ -106,33 +75,45 @@ explain_decompression(Query *query, int cursorOptions, IntoClause *into, Explain
 	standard_ExplainOneQuery(query, cursorOptions, into, es, queryString, params, queryEnv);
 	if (decompress_cache_print)
 	{
-		Assert(es->format < sizeof(format_texts) / sizeof(*format_texts));
-
-		ExplainOpenGroup("Array cache", "Arrow Array Cache", true, es);
-		ExplainPropertyInteger(format_texts[es->format].hits_text,
-							   NULL,
-							   decompress_cache_stats.hits,
-							   es);
-		ExplainPropertyInteger(format_texts[es->format].miss_text,
-							   NULL,
-							   decompress_cache_stats.misses,
-							   es);
-		ExplainPropertyInteger(format_texts[es->format].evict_text,
-							   NULL,
-							   decompress_cache_stats.evictions,
-							   es);
-		ExplainPropertyInteger(format_texts[es->format].decompress_text,
-							   NULL,
-							   decompress_cache_stats.decompressions,
-							   es);
-
-		if (es->verbose)
-			ExplainPropertyInteger(format_texts[es->format].decompress_calls_text,
-								   NULL,
-								   decompress_cache_stats.decompress_calls,
-								   es);
-
-		ExplainCloseGroup("Array cache", "Arrow Array Cache", true, es);
+		const bool has_decompress_data = decompress_cache_stats.decompressions > 0 ||
+										 decompress_cache_stats.decompress_calls > 0;
+		const bool has_cache_data = decompress_cache_stats.hits > 0 ||
+									decompress_cache_stats.misses > 0 ||
+									decompress_cache_stats.evictions > 0;
+		if (has_decompress_data || has_cache_data)
+		{
+			if (es->format == EXPLAIN_FORMAT_TEXT)
+			{
+				appendStringInfoString(es->str, "Array:");
+				if (has_cache_data)
+					appendStringInfoString(es->str, " cache");
+				APPEND_IF_POSITIVE(es->str, "hits=%lld", decompress_cache_stats.hits);
+				APPEND_IF_POSITIVE(es->str, "misses=%lld", decompress_cache_stats.misses);
+				APPEND_IF_POSITIVE(es->str, "evictions=%lld", decompress_cache_stats.evictions);
+				if (has_decompress_data)
+					appendStringInfoString(es->str, ", decompress");
+				APPEND_IF_POSITIVE(es->str, "count=%lld", decompress_cache_stats.decompressions);
+				APPEND_IF_POSITIVE(es->str, "calls=%lld", decompress_cache_stats.decompress_calls);
+				appendStringInfoChar(es->str, '\n');
+			}
+			else
+			{
+				ExplainOpenGroup("Array Cache", "Arrow Array Cache", true, es);
+				ExplainPropertyInteger("hits", NULL, decompress_cache_stats.hits, es);
+				ExplainPropertyInteger("misses", NULL, decompress_cache_stats.misses, es);
+				ExplainPropertyInteger("evictions", NULL, decompress_cache_stats.evictions, es);
+				ExplainCloseGroup("Array Cache", "Arrow Array Cache", true, es);
+
+				ExplainOpenGroup("Array Decompress", "Arrow Array Decompress", true, es);
+				ExplainPropertyInteger("count", NULL, decompress_cache_stats.decompressions, es);
+				if (es->verbose)
+					ExplainPropertyInteger("calls",
+										   NULL,
+										   decompress_cache_stats.decompress_calls,
+										   es);
+				ExplainCloseGroup("Array Decompress", "Arrow Array Decompress", true, es);
+			}
```

Comment:
We need to add tests for other EXPLAIN output format than TEXT

Reply:
Looks like we also need to test using `VERBOSE` on the explain to fully cover the code paths

Reply:
There is no reason "calls" should be only for verbose output, so removed this check.